test/Microsoft.AspNetCore.Diagnostics.Tests/ExceptionHandlerTest.cs

```diff
@@ -388,5 +481,26 @@ public class ExceptionHandlerTest
             Assert.NotNull(listener.DiagnosticHandledException?.HttpContext);
             Assert.NotNull(listener.DiagnosticHandledException?.Exception);
         }
+
+        public class InspectAfterReExecuteMiddleware
+        {
+            public const string QueryStringDifferenceHeader = "QUERYSTRINGDIFFERENT";
```

Comment:
I had been trying to return the success or failure in the header but realized I could just do the assert in the Middleware, missed it in cleanup.